perfkitbenchmarker/alicloud/ali_virtual_machine.py

```diff
@@ -152,10 +234,19 @@ def _PostCreate(self):
     stdout, _ = vm_util.IssueRetryableCommand(describe_cmd)
     response = json.loads(stdout)
     instance = response['Instances']['Instance'][0]
-    assert instance['PublicIpAddress']['IpAddress'][0] == self.ip_address
-    self.internal_ip = instance['InnerIpAddress']['IpAddress'][0]
+    if self.network.use_vpc:
+      pub_ip_address = instance['EipAddress']['IpAddress']
+      self.internal_ip = \
+          instance['VpcAttributes']['PrivateIpAddress']['IpAddress'][0]
+    else:
+      pub_ip_address = instance['PublicIpAddress']['IpAddress'][0]
+      self.internal_ip = instance['InnerIpAddress']['IpAddress'][0]
+    assert self.ip_address == pub_ip_address
     self.group_id = instance['SecurityGroupIds']['SecurityGroupId'][0]
 
+    self._WaitForInstanceStatus(['Running'])
+    self.bootable_time = time.time()
```

Comment:
This should only be set after we've successfully sshed into the VM - it should get set for you at the appropriate time, so you don't need this line